services/localfile/client/utils.go

```diff
@@ -13,28 +13,31 @@ import (
 	"github.com/Snowflake-Labs/sansshell/services/util"
 )
 
-// ReadRemoteFile is a helper function for reading a single file from a remote host
-// using a proxy.Conn. If the conn is defined for >1 targets this will return an error.
-func ReadRemoteFile(ctx context.Context, conn *proxy.Conn, path string) ([]byte, error) {
-	if len(conn.Targets) != 1 {
-		return nil, errors.New("ReadRemoteFile only supports single targets")
-	}
+type ReadRemoteFileResponse struct {
+	Target   string
+	Index    int
+	Contents []byte
+}
 
-	c := pb.NewLocalFileClient(conn)
-	stream, err := c.Read(ctx, &pb.ReadActionRequest{
+// ReadRemoteFile is a helper function for reading a single file from a remote host using a proxy.Conn.
```

Comment:
This is part of our public API (even though it's buried a bit deeply), so I'd like to avoid unnecessarily changing the function signature in backwards compatible ways. What do you think of adding a new function like ReadRemoteFileFromMany and making ReadRemoteFile a small wrapper around it?

Reply:
Great point! I'll take that suggestion

Reply:
Done